File: programers_Lv1/최대공약수와 최소공배수.js
// https://school.programmers.co.kr/learn/courses/30/lessons/12940?language=javascript
function solution(n, m) {
  // 최대 공약수 구하기
  const gcdFunc = (a, b) => (a % b === 0 ? b : gcdFunc(b, a % b));
  const gcd = gcdFunc(n, m);

  // 최소 공배수 구하기
  const lcm = Number((n * m) / gcd);

  return [gcd, lcm];
}

console.log(solution(3, 12));

File: programers_Lv1/최대공약수와 최소공배수.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12940?language=python3
def solution(n, m):
    answer = []

    def gcd_func(a, b):
        return b if a % b == 0 else gcd_func(b, a % b)

    # 최소공배수 = 두 자연수의 곱 / 최대공약수
    gcd = gcd_func(n, m)
    lcm = int(n * m / gcd)

    return [gcd, lcm]


print(solution(3, 12))
